### PR TITLE
fix promise handling in remote object error case

### PR DESCRIPTION
**Pull request by nsaenz7**
the rollback functionality wasn't being triggered correctly due to an incorrect promise chain. fix this, and add a test to verify that if an error happens in remote doc upload, we are rolling back both the remote change and sql change(s).

### CODE REVIEW

File: components/persistor/lib/knex/update.ts [hunk 1]
```diff
@@ -216,7 +216,9 @@ module.exports = function (PersistObjectTemplate) {
         promises.push(this.saveKnexPojo(obj, pojo, isDocumentUpdate ? obj._id : null, txn, logger));
 
         return Promise.all(promises) // update sql saves first
-            .then(Promise.all(remoteUpdates)) // update remote objects second
+            .then(() => {
```
Comment (zamn) [inline components/persistor/lib/knex/update.ts:219]: this can just be
```javascript
.then(() => Promise.all(remoteUpdates))
```

**Review by sshankar19 — APPROVED** [review]

File: components/persistor/lib/knex/update.ts (hunk 1 above)
Comment (nsaenz7) [reply components/persistor/lib/knex/update.ts:219]: I like the spirit of this! for the sake of time to make the small change but then retest (yes we do have unit tests, but also I did manual testing on the current branch with actual S3 connection), I'm gonna say this is probably OK as is

**Review by jacoblb64 — APPROVED** [review]
Looks great to me, love the sinon mocking

**Comment by nsaenz7** [comment]
@jacoblb64 using sinon here was @neaket 's idea! it looks like we already had sinon as a dep but perhaps could use it even more in the future as we add more detailed test cases, especially around S3 functionality and mocking the different failure cases.

**Pull request merged**